### PR TITLE
Add gRPC code to the Ballerina in Action Section

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Add the following gRPC sample.

```proto
syntax = "proto3";

import "google/protobuf/wrappers.proto";
import "google/protobuf/empty.proto";

service Store {
  rpc storeAlbum(Album) returns (Album);
  rpc getAlbumFromId(google.protobuf.StringValue) returns (Album);
  rpc getAlbums(google.protobuf.Empty) returns (stream Album);
}

message Album {
    string id = 1;
    string title = 2;
    string artist = 3;
    float price = 4;
};
```

```ballerina
import ballerina/grpc;

listener grpc:Listener ep = new (9090);

type ReadOnlyAlbum readonly & record {|
    string id;
    string title;
    string artist;
    float price;
|};

table<ReadOnlyAlbum> key(id) albums = table [
        {id: "1", title: "Blue Train", artist: "John Coltrane", price: 56.99},
        {id: "2", title: "Jeru", artist: "Gerry Mulligan", price: 17.99},
        {id: "3", title: "Sarah Vaughan and Clifford Brown", artist: "Sarah Vaughan", price: 39.99}
    ];

@grpc:ServiceDescriptor {descriptor: ROOT_DESCRIPTOR_RECORD_STORE, descMap: getDescriptorMapRecordStore()}
service "RecordStore" on ep {

    remote function getAlbumFromId(string id) returns Album|error {
        ReadOnlyAlbum? album = albums[id];
        if album is ReadOnlyAlbum {
            return <Album>album;
        }
        return error grpc:NotFoundError(string `Related album for ID ${id} not found`);
    }

    remote function storeAlbum(Album album) returns Album|error {
        albums.add(album.cloneReadOnly());
        return album;
    }

    remote function getAlbums() returns stream<Album, error?>|error {
        return albums.toArray().toStream();
    }
}
```

## Sample View 
Screenshot 2022-01-31 at 17 48 14

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -352,6 +353,57 @@ <h2>Ballerina in Action</h2>
...
+      }
+}                                 
+                              </code>
+                           </pre>
+                        </div>
+                     </div>
+                  </div>
+               </div>
+            </div>
          </div>
```
Comment (dilanSachi) [inline index.html:407]: are we going to add the proto def separately?

Comment (BuddhiWathsala) [reply index.html:437]: Addressed in https://github.com/ballerina-platform/ballerina-dev-website/pull/3692/commits/26c60765bd595915df9aac7ffc3d0cf7334e532f

**Comment by daneshk** [comment]
@BuddhiWathsala shall we rename `getAlbumFromId` to `getAlbumById`

**Comment by daneshk** [comment]
Rather than having `ReadonlyAlbum` type, can we can the generated `Album` type

**Comment by daneshk** [comment]
> > getAlbumById
> 
> Did you mean to reuse the generated `Album`?

Yes, can we use it

**Review by shafreenAnfar — APPROVED** [review]

**Pull request closed without merging**